zio-http/src/test/scala/zhttp/service/ClientSpec.scala

```diff
@@ -43,6 +43,19 @@ object ClientSpec extends HttpRunnableSpec {
       test("handle connection failure") {
         val res = Client.request("http://localhost:1").either
         assertM(res)(isLeft(isSubtype[ConnectException](anything)))
+      } +
+      test("handle proxy connection failure") {
+        val res =
+          for {
+            validServerPort <- ZIO.service[DynamicServer].flatMap(_.port)
+            serverUrl       <- ZIO.fromEither(URL.fromString(s"http://localhost:$validServerPort"))
+            proxyUrl        <- ZIO.fromEither(URL.fromString("http://localhost:0001"))
+            out             <- Client.request(
+              Request(url = serverUrl),
+              Config().withProxy(Proxy(proxyUrl)),
+            )
+          } yield out
+        assertM(res.either)(isLeft(isSubtype[ConnectException](anything)))
```

Comment:
Can we add a test for success also?

Reply:
Hi!
Thanks for the feedback
I've fixed all comments above 😌